fuel/transformers/__init__.py

```diff
@@ -499,13 +602,34 @@ class Merge(Transformer):
     ('Hello world!', 'Bonjour le monde!')
 
     """
-    def __init__(self, data_streams, sources):
+    def __init__(self, data_streams, sources, axis_labels=None):
+        super(Merge, self).__init__(
+            iteration_scheme=None, axis_labels=axis_labels)
+        if not all(data_stream.produces_examples ==
+                   data_streams[0].produces_examples
+                   for data_stream in data_streams):
+            raise ValueError('all data streams must produce the same type of '
+                             'output (batches or examples)')
         self.data_streams = data_streams
+        self.produces_examples = self.data_streams[0].produces_examples
+
         if len(list(chain(*[data_stream.sources for data_stream
                             in data_streams]))) != len(sources):
             raise ValueError("wrong number of sources given")
         self.sources = sources
 
+    def close(self):
```

Comment:
See my reply about ADS. `Merge` has a list of data streams, so the default implementation in `Transformer` would crash.